src/PointwiseFunctions/GeneralRelativity/ComputeSpacetimeQuantities.cpp

```diff
@@ -257,14 +286,25 @@ template <size_t SpatialDim, typename Frame, typename DataType>
 tnsr::A<DataType, SpatialDim, Frame> spacetime_normal_vector(
     const Scalar<DataType>& lapse,
     const tnsr::I<DataType, SpatialDim, Frame>& shift) noexcept {
-  auto spacetime_normal_vector =
-      make_with_value<tnsr::A<DataType, SpatialDim, Frame>>(lapse, 0.);
-  get<0>(spacetime_normal_vector) = 1. / get(lapse);
+  tnsr::A<DataType, SpatialDim, Frame> local_spacetime_normal_vector{
+      get_size(get(lapse))};
+  spacetime_normal_vector(make_not_null(&local_spacetime_normal_vector), lapse,
+                          shift);
+  return local_spacetime_normal_vector;
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+void spacetime_normal_vector(
+    const gsl::not_null<tnsr::A<DataType, SpatialDim, Frame>*>
+        spacetime_normal_vector,
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift) noexcept {
+  destructive_resize_components(spacetime_normal_vector, get_size(get(lapse)));
```

Comment:
I was under the impression that the `UNLIKELY` resize was generally preferable to `ASSERT`s in these pass-by-pointer free functions. Has there been a change in attitude, or is there a more subtle case-by-case rule?

Reply:
I don't think the `not_null` functions should ever change the size, only the values. By changing where the allocation is it basically means it's impossible to use Variables and any classes like it safely...

Reply:
Am I incorrect that we have done this deliberately in the past for the generalized harmonic computations?

Reply:
It does appear to be what is done in the GH system. I do not remember how/when it was decided to be done like this.

Reply:
It appears to have first been a suggestion by @kidder to the general practice. The first record I can find is here : https://github.com/sxs-collaboration/spectre/pull/1177#discussion_r239847742 .

Reply:
Yea... But I think it's bad because it doesn't change the value of the pointer but what it's pointing to if one regards the data as the fundamental part. I think if we allow the `not_null` functions to change the sizes then we cannot use anything like Variables anymore because you can call a function and all of a sudden nothing is right anymore... It's basically just asking for horrid memory problems that I don't know how I would debug except for sticking print statements in everywhere... I can chat with @kidder and @wthrowe about this tomorrow :)

Reply:
I think the current implementation should error when applied to a variables that's the wrong size, because it would destructively resize a non-owning `VectorImpl`, but I can test that.

Reply:
Ah okay! That's my only concern. Basically resizing non-owning `*Vectors` should be not allowed :)